src/js/components/productSupplier/modals/PreferenceTypeModal.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+import { RiCloseFill } from 'react-icons/all';
+import Modal from 'react-modal';
+
+import Button from 'components/form-elements/Button';
+import { PRODUCT_SUPPLIER_URL } from 'consts/applicationUrls';
+import Translate from 'utils/Translate';
+
+const PreferenceTypeModal = ({
+  isOpen,
+  closeModal,
+  modalData,
+  productSupplierId,
+}) => {
+  const mapPreferenceTypes = (preferenceTypes) => {
+    const defaultPreferenceTypePlaceholder = (
+      <p className="default-preference-type">
+        <span className="preference-type-location">Default: </span>
+      </p>
+    );
+
+    return preferenceTypes?.reduce((acc, preferenceType) => {
+      const {
+        preferenceType: { name },
+        destinationParty,
+      } = preferenceType;
+
+      if (!destinationParty) {
+        return {
+          ...acc,
+          default: (
+            <p className="default-preference-type">
+              <span className="preference-type-location">Default: </span>
+              <span>{name}</span>
+            </p>
+          ),
+        };
+      }
+
+      return {
+        ...acc,
+        preferenceTypes: [
+          ...acc.preferenceTypes,
+          <p>
+            <span className="preference-type-location">
+              {destinationParty?.name}
+              :
+              {' '}
+            </span>
+            <span>{name}</span>
+          </p>,
+        ],
+      };
+    }, { preferenceTypes: [], default: defaultPreferenceTypePlaceholder });
+  };
+
+  const redirectToEditProductSource = (id) => {
+    window.location = PRODUCT_SUPPLIER_URL.edit(id);
+  };
+
+  return (
+    <Modal isOpen={isOpen} className="modal-content">
+      <div>
+        <div className="d-flex justify-content-between">
+          <p className="preference-type-modal-header">
+            <Translate
+              id="react.productSupplier.column.preferenceType.label"
+              defaultMessage="Preference Type"
+            />
+          </p>
+          <RiCloseFill
+            size="32px"
+            className="cursor-pointer"
+            role="button"
+            onClick={closeModal}
+          />
+        </div>
+        <div className="preference-type-modal-list-container">
+          <div>
+            {mapPreferenceTypes(modalData).default}
```

Comment:
let's assign the result of the `mapPreferenceTypes(modalData)` to a variable, because doing it this way, you run this method twice - for the first time here, and for the second time in the 85th line.